pest/src/parser_state.rs

```diff
@@ -433,6 +433,19 @@ impl<'i, R: RuleType> ParserState<'i, R> {
         }
     }
 
+    /// Asks the `ParserState` to match a single character based on a filter function.
+    /// If the match is successful, this will return an `Ok` with the updated `Box<ParserState>`.
+    /// If failed, an `Err` with the updated `Box<ParserState>` is returned.
```

Comment:
Should probably include an example here.

Reply:
I've added an example using `char::is_ascii`. It could be implemented better as a builtin by just checking the first byte, but as an example it's easy to understand.